src/app/shared/components/template/services/template-nav.service.ts

```diff
@@ -79,11 +79,16 @@ export class TemplateNavService extends SyncServiceBase {
   public handleNavAction(action: FlowTypes.TemplateRowAction) {
     // TODO: Find more elegant way to get current root level template name
     const parentName = location.pathname.replace("/template/", "");
-    const [templatename] = action.args;
+    const [templatename, key, value] = action.args;
     const nav_parent_triggered_by = action._triggeredBy?.name;
     const queryParams: INavQueryParams = { nav_parent: parentName, nav_parent_triggered_by };
     // handle direct page or template navigation
     const navTarget = templatename.startsWith("/") ? [templatename] : ["template", templatename];
+
+    // If "dismiss_on_return" is set to true for the go_to action, dismiss the current popup before navigating away
+    if (key === "dismiss_on_return" && parseBoolean(value)) {
+      history.replaceState({}, "", location.pathname);
```

Comment:
Originally I tried using `this.location.go(location.pathname)` and `router.navigate()`, but in that case the page with the query params defining the popup remained in the history, meaning that upon return to the parent template, navigating back would cause the popup to reappear.

Reply:
Like the idea of allowing authors to explicitly pass a param to manage this instead of having to rely tracking weird nav history. Although I'm not sure if this line would inadvertently remove additional url info that we might want to keep. 

An alternative snippet that I've just tested locally (and appears to work) might be:
```ts
 if (key === "dismiss_on_return" && parseBoolean(value)) {
      const { popup_child } = this.route.snapshot.queryParams;
      if (popup_child) {
        const popupDismissParams: INavQueryParams = {
          popup_child: null,
          popup_parent: null,
          popup_parent_triggered_by: null,
        };
        // alter route history so that on back-nav popup will not be present
        await this.router.navigate([], {
          relativeTo: this.route,
          queryParams: popupDismissParams,
          queryParamsHandling: "merge",
          replaceUrl: true,
        });
        // Dismiss open popup (without await to allow rest of nav to proceed await)
        this.dismissPopup(popup_child);
      }
    }
```

However as the PR is already merged I'll leave for now and you can revisit if issue comes up